src/System.Private.CoreLib/shared/System/Environment.Unix.cs

```diff
@@ -444,5 +444,24 @@ private static int CheckedSysConf(Interop.Sys.SysConfName name)
             }
             return (int)result;
         }
+
+        public static long WorkingSet
+        {
+            get
+            {
+                Type? processType = Type.GetType("System.Diagnostics.Process, System.Diagnostics.Process, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);
+                if (processType?.GetMethod("GetCurrentProcess")?.Invoke(null, BindingFlags.DoNotWrapExceptions, null, null, null) is IDisposable currentProcess)
```

Comment:
Do we need to fix the perf of this one too? How much more expensive is this compared to Windows? Does the cost of this show up in the profiles when you turn on performance counters that call this again and again?

Reply:
Ideally, yes. But my benchmarks using BenchmarkDotNet suggested that Linux is at a better state than Windows (>10x) so for now I'm fixing Windows first. 

Linux:
```
|     Method |     N |     Mean |    Error |   StdDev | Rank |
|----------- |------ |---------:|---------:|---------:|-----:|
| workingset | 10000 | 143.4 us | 2.520 us | 2.801 us |    1 |
```

Windows (before the fix):
```
|     Method |     N |     Mean |     Error |    StdDev | Rank |
|----------- |------ |---------:|----------:|----------:|-----:|
| workingset | 10000 | 1.620 ms | 0.0153 ms | 0.0127 ms |    1 |
```

In addition to this, the customer requesting this fix is Bing and they are running on Windows.

That being said, the Linux overhead is also huge compared to what we have on Windows after this fix so ideally we'd like to fix it on Linux too: 
```
|     Method |     N |     Mean |     Error |    StdDev |   Median | Rank |
|----------- |------ |---------:|----------:|----------:|---------:|-----:|
| workingset | 10000 | 851.6 ns | 16.916 ns | 34.556 ns | 835.9 ns |    1 |
```